DataQualityTester/views/quality.py

```diff
@@ -143,19 +143,19 @@ def download_results(task_id):
     test_set_id = request.args.get('test_set')
     test_set = TestSet(test_set_id)
     csv_file = csv.writer(csv_response)
-    csv_file.writerow(('type', 'name', 'score', 'total_tested', 'failed', 'passed', 'not-relevant'))
+    csv_file.writerow(('type', 'indicator_num', 'name', 'score', 'total_tested', 'failed', 'passed', 'not-relevant'))
     for component in test_set.components:
         component_file = '{}.json'.format(join(output_path, component.id))
         with open(component_file) as f:
             results = json.load(f)
             category_score = _compute_score(results)
-        csv_file.writerow(['category', component.name, category_score])
+        csv_file.writerow(['category', '', component.name, category_score])
 
         for test, test_scores in results.items():
             test_score = percent(test_scores)
-            print(test)
+            indicator = INDICATOR_LOOKUP[test]
```

Comment:
Getting a key error on some results for this dict.

I used a try/catch block as we know this lookup can fail. 
https://github.com/pwyf/data-quality-tester/blob/2021-develop/DataQualityTester/views/quality.py#L67

Might be worth abstracting that into a "get_indicator_num" and putting the catch in there.. 
Perhaps there is also one title missing from the spreadsheet/different in the spreadsheet from the test.

```
Traceback (most recent call last):
  File "/home/michael/dev/ods/pwyf/data-quality-tester/venv/lib/python3.6/site-packages/flask/app.py", line 2309, in __call__
    return self.wsgi_app(environ, start_response)
  File "/home/michael/dev/ods/pwyf/data-quality-tester/venv/lib/python3.6/site-packages/flask/app.py", line 2295, in wsgi_app
    response = self.handle_exception(e)
  File "/home/michael/dev/ods/pwyf/data-quality-tester/venv/lib/python3.6/site-packages/flask/app.py", line 1741, in handle_exception
    reraise(exc_type, exc_value, tb)
  File "/home/michael/dev/ods/pwyf/data-quality-tester/venv/lib/python3.6/site-packages/flask/_compat.py", line 35, in reraise
    raise value
  File "/home/michael/dev/ods/pwyf/data-quality-tester/venv/lib/python3.6/site-packages/flask/app.py", line 2292, in wsgi_app
    response = self.full_dispatch_request()
  File "/home/michael/dev/ods/pwyf/data-quality-tester/venv/lib/python3.6/site-packages/flask/app.py", line 1815, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "/home/michael/dev/ods/pwyf/data-quality-tester/venv/lib/python3.6/site-packages/flask/app.py", line 1718, in handle_user_exception
    reraise(exc_type, exc_value, tb)
  File "/home/michael/dev/ods/pwyf/data-quality-tester/venv/lib/python3.6/site-packages/flask/_compat.py", line 35, in reraise
    raise value
  File "/home/michael/dev/ods/pwyf/data-quality-tester/venv/lib/python3.6/site-packages/flask/app.py", line 1813, in full_dispatch_request
    rv = self.dispatch_request()
  File "/home/michael/dev/ods/pwyf/data-quality-tester/venv/lib/python3.6/site-packages/flask/app.py", line 1799, in dispatch_request
    return self.view_functions[rule.endpoint](**req.view_args)
  File "/home/michael/dev/ods/pwyf/data-quality-tester/DataQualityTester/routes.py", line 52, in download_results
    response_csv = quality.download_results(uuid)
  File "/home/michael/dev/ods/pwyf/data-quality-tester/DataQualityTester/views/quality.py", line 156, in download_results
    indicator = INDICATOR_LOOKUP[test]
KeyError: 'Country strategy or Memorandum of Understanding'
```

Reply:
Thanks, I didn't get that error, I wonder if the indicator definitions are different. I've added a try/catch block as per your previous changes.